### PR TITLE
Fix `PurchasesIntegrationTest` building issue (#996 into main)

### PR DESCRIPTION
**Pull request by vegaro**
Merging #996 into main

### CODE REVIEW

**Comment by vegaro** [comment]
Not sure what's going on. One of the tests is failing. I don't have access to the Firebase project right now so I can't see what the failure is.

**Comment by pablo-guardiola** [comment]
```
1 test failed  out of 5

canPurchaseSubsProduct

com.revenuecat.purchases.PurchasesIntegrationTest

java.lang.AssertionError: Verification failed: call 1 of 1: BillingAbstract(#1).makePurchaseAsync(any(), eq(android-integration-test-1683246465943), matcher<PurchasingData>(), null(), null(), eq(false))). Only one matching call to BillingAbstract(#1)/makePurchaseAsync(Activity, String, PurchasingData, ReplaceProductInfo, String, Boolean) happened, but arguments are not matching:
[0]: argument: com.revenuecat.purchases.MainActivity@bcbcf58, matcher: any(), result: +
[1]: argument: android-integration-test-1683246465943, matcher: eq(android-integration-test-1683246465943), result: +
[2]: argument: Subscription(productId=monthly_intro_pricing_one_week, optionId=p1m, productDetails=ProductDetails(#4), token=test-offer-token), matcher: matcher<PurchasingData>(), result: +
[3]: argument: null, matcher: null(), result: +
[4]: argument: null, matcher: null(), result: +
[5]: argument: null, matcher: eq(false), result: -

Stack trace:
                                             io.mockk.impl.InternalPlatform.captureStackTrace                                           (InternalPlatform.kt:125)             
                                               io.mockk.impl.stub.MockKStub.handleInvocation                                            (MockKStub.kt:255)                    
             io.mockk.impl.instantiation.JvmMockFactoryHelper$mockHandler$1.invocation                                                  (JvmMockFactoryHelper.kt:24)          
                               io.mockk.proxy.common.ProxyInvocationHandler.invoke                                                      (ProxyInvocationHandler.kt:36)        
                    com.revenuecat.purchases.common.BillingAbstract_1_Proxy.isConnected                                                 (-:13)                                
                                         com.revenuecat.purchases.Purchases.updatePendingPurchaseQueue$purchases_integrationTestRelease (Purchases.kt:1684)                   
                                         com.revenuecat.purchases.Purchases.onAppForegrounded                                           (Purchases.kt:212)                    
                               com.revenuecat.purchases.AppLifecycleHandler.onStart                                                     (AppLifecycleHandler.kt:11)           
                            androidx.lifecycle.FullLifecycleObserverAdapter.onStateChanged                                              (FullLifecycleObserverAdapter.java:39)
                     androidx.lifecycle.LifecycleRegistry$ObserverWithState.dispatchEvent                                               (LifecycleRegistry.java:360)          
                                       androidx.lifecycle.LifecycleRegistry.addObserver                                                 (LifecycleRegistry.java:202)          
                                       com.revenuecat.purchases.Purchases$1.invoke                                                      (Purchases.kt:164)                    
                                       com.revenuecat.purchases.Purchases$1.invoke                                                      (Purchases.kt:163)                    
                                         com.revenuecat.purchases.Purchases.dispatch                                                    (Purchases.kt:1352)                   
                                         com.revenuecat.purchases.Purchases.<init>                                                      (Purchases.kt:163)                    
                                         com.revenuecat.purchases.Purchases.<init>                                                      (Purchases.kt:84)                     
                                  com.revenuecat.purchases.PurchasesFactory.createPurchases                                             (PurchasesFactory.kt:165)             
                             com.revenuecat.purchases.PurchasesExtensionsKt.configure                                                   (purchasesExtensions.kt:10)           
                  com.revenuecat.purchases.PurchasesIntegrationTest$setup$1.invoke                                                      (PurchasesIntegrationTest.kt:80)      
                  com.revenuecat.purchases.PurchasesIntegrationTest$setup$1.invoke                                                      (PurchasesIntegrationTest.kt:68)      
                          com.revenuecat.purchases.PurchasesIntegrationTest.onActivityReady$lambda-4                                    (PurchasesIntegrationTest.kt:221)     
                          com.revenuecat.purchases.PurchasesIntegrationTest.$r8$lambda$skmKbmN6QaNJDsyjNQYOu78jXxM                      (-:0)                                 
com.revenuecat.purchases.PurchasesIntegrationTest$$ExternalSyntheticLambda1.perform                                                     (-:4)                                 
                                    androidx.test.core.app.ActivityScenario.lambda$onActivity$2$ActivityScenario                        (ActivityScenario.java:660)           
                          androidx.test.core.app.ActivityScenario$$Lambda$4.run                                                         (ActivityScenario.java:652)           
                             java.util.concurrent.Executors$RunnableAdapter.call                                                        (Executors.java:462)                  
                                            java.util.concurrent.FutureTask.run                                                         (FutureTask.java:266)                 
                                   android.app.Instrumentation$SyncRunnable.run                                                         (Instrumentation.java:2223)           
                                                         android.os.Handler.handleCallback                                              (Handler.java:938)                    
                                                         android.os.Handler.dispatchMessage                                             (Handler.java:99)                     
                                                          android.os.Looper.loop                                                        (Looper.java:223)                     
                                                 android.app.ActivityThread.main                                                        (ActivityThread.java:7664)            
                                                   java.lang.reflect.Method.invoke                                                      (Method.java:-2)N                     
                    com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run                                                         (RuntimeInit.java:592)                
                                         com.android.internal.os.ZygoteInit.main                                                        (ZygoteInit.java:947)                 
	at io.mockk.impl.recording.states.VerifyingState.failIfNotPassed(VerifyingState.kt:63)
	at io.mockk.impl.recording.states.VerifyingState.recordingDone(VerifyingState.kt:42)
	at io.mockk.impl.recording.CommonCallRecorder.done(CommonCallRecorder.kt:47)
	at io.mockk.impl.eval.RecordedBlockEvaluator.record(RecordedBlockEvaluator.kt:64)
	at io.mockk.impl.eval.VerifyBlockEvaluator.verify(VerifyBlockEvaluator.kt:30)
	at io.mockk.MockKDsl.internalVerify(API.kt:119)
	at io.mockk.MockKKt.verify(MockK.kt:149)
	at io.mockk.MockKKt.verify$default(MockK.kt:140)
	at com.revenuecat.purchases.PurchasesIntegrationTest
```

**Comment by pablo-guardiola** [comment]
It seems that https://github.com/RevenueCat/purchases-android/blob/1b2095c364f83c93d629ab935a82d7c733a0fec0/purchases/src/androidTestIntegrationTest/kotlin/com/revenuecat/purchases/PurchasesIntegrationTest.kt#L211 is `true`. Did something change on the backend recently?

**Comment by tonidero** [comment]
This was fixed in #980 which was merged just today. If you rebase, this should work

File: integration-tests/build.gradle [hunk 1]
```diff
@@ -47,7 +47,6 @@ dependencies {
     androidTestImplementation 'androidx.test:runner:1.4.0'
     androidTestImplementation 'androidx.test:rules:1.4.0'
     androidTestImplementation 'androidx.test.ext:junit-ktx:1.1.3'
-    androidTestImplementation 'org.assertj:assertj-core:3.22.0'
```
Comment (tonidero) [inline integration-tests/build.gradle:50]: Hmm interesting that R8 is failing with this, maybe we could also try to update to the latest assertj version to see if they fixed it. Haven't found the CHANGELOG though, so might not fix it...

Comment (vegaro) [reply integration-tests/build.gradle:50]: Oh AS wasn't offering me to upgrade to a more recent version so I assumed this was the latest one, let me try with 3.24.2

Comment (vegaro) [reply integration-tests/build.gradle:50]: Ok, tried and it still failed [`22740ec` (#998)](https://github.com/RevenueCat/purchases-android/pull/998/commits/22740eca8d337a9a5f640682b54c1132e3b04ba2)

**Review by tonidero — APPROVED** [review]

**Review by tonidero — APPROVED** [review]

**Review by tonidero — APPROVED** [review]

**Review by tonidero — APPROVED** [review]

**Comment by tonidero** [comment]
Lol, github was stuck so I retried a few times 😅

**Comment by tonidero** [comment]
@vegaro Is this ready to merge?

**Pull request merged**